src/lint.cpp

```diff
@@ -327,6 +333,12 @@ void linter::declare_variable(scope &scope, identifier name, variable_kind kind,
                    declared, used_var.name,
                    /*is_assigned_before_declaration=*/false);
                break;
+             case used_variable_kind::use_and_assignment:
+               this->report_error_if_assignment_is_illegal(
+                   declared, used_var.name,
+                   /*is_assigned_before_declaration=*/false,
+                   /* is_used_before_declaration=*/true);
```

Comment:
Minor: `is_assigned_before_declaration=false is_used_before_declaration=true` doesn't make sense. It behaves the same as `is_assigned_before_declaration=false is_used_before_declaration=false`. Perhaps the boolean parameter should be an enum instead?

Reply:
I think adding an enum would be better, in addition to facilitating the addition of new cases if any, and reduces the number of parameters. Does it add to lint.h?